File: examples/polydiff/PolyLexer.php
<?php
// $ANTLR 3.1.3 ??? 05, 2009 23:29:28 Poly.g 2009-05-06 00:14:36


# for convenience in actions
if (!defined('HIDDEN')) define('HIDDEN', BaseRecognizer::$HIDDEN);

      
 

class PolyLexer extends AntlrLexer {
    static $WS=7;
    static $INT=5;
    static $MULT=4;
    static $ID=6;
    static $EOF=-1;
    static $T__9=9;
    static $T__8=8;

    // delegates
    // delegators

    function __construct($input, $state=null){
        parent::__construct($input,$state);

        
    }
    function getGrammarFileName() { return "Poly.g"; }

    // $ANTLR start "T__8"
    function mT__8(){
        try {
            $_type = PolyLexer::$T__8;
            $_channel = PolyLexer::$DEFAULT_TOKEN_CHANNEL;
            // Poly.g:7:6: ( '+' ) 
            // Poly.g:7:8: '+' 
            {
            $this->matchChar(43); 

            }

            $this->state->type = $_type;
            $this->state->channel = $_channel;
        }
        catch(Exception $e){
            throw $e;
        }
    }
    // $ANTLR end "T__8"

    // $ANTLR start "T__9"
    function mT__9(){
        try {
            $_type = PolyLexer::$T__9;
            $_channel = PolyLexer::$DEFAULT_TOKEN_CHANNEL;
            // Poly.g:8:6: ( '^' ) 
            // Poly.g:8:8: '^' 
            {
            $this->matchChar(94); 

            }

            $this->state->type = $_type;
            $this->state->channel = $_channel;
        }
        catch(Exception $e){
            throw $e;
        }
    }
    // $ANTLR end "T__9"

    // $ANTLR start "ID"
    function mID(){
        try {
            $_type = PolyLexer::$ID;
            $_channel = PolyLexer::$DEFAULT_TOKEN_CHANNEL;
            // Poly.g:21:4: ( ( 'a' .. 'z' )+ ) 
            // Poly.g:21:6: ( 'a' .. 'z' )+ 
            {
            // Poly.g:21:6: ( 'a' .. 'z' )+ 
            $cnt1=0;
            //loop1:
            do {
                $alt1=2;
                $LA1_0 = $this->input->LA(1);

                if ( (($LA1_0>=$this->getToken('97') && $LA1_0<=$this->getToken('122'))) ) {
                    $alt1=1;
                }


                switch ($alt1) {
            	case 1 :
            	    // Poly.g:21:6: 'a' .. 'z' 
            	    {
            	    $this->matchRange(97,122); 

            	    }
            	    break;

            	default :
            	    if ( $cnt1 >= 1 ) break 2;//loop1;
                        $eee =
                            new EarlyExitException(1, $this->input);
                        throw $eee;
                }
                $cnt1++;
            } while (true);


            }

            $this->state->type = $_type;
            $this->state->channel = $_channel;
        }
        catch(Exception $e){
            throw $e;
        }
    }
    // $ANTLR end "ID"

    // $ANTLR start "INT"
    function mINT(){
        try {
            $_type = PolyLexer::$INT;
            $_channel = PolyLexer::$DEFAULT_TOKEN_CHANNEL;
            // Poly.g:23:5: ( ( '0' .. '9' )+ ) 
            // Poly.g:23:7: ( '0' .. '9' )+ 
            {
            // Poly.g:23:7: ( '0' .. '9' )+ 
            $cnt2=0;
            //loop2:
            do {
                $alt2=2;
                $LA2_0 = $this->input->LA(1);

                if ( (($LA2_0>=$this->getToken('48') && $LA2_0<=$this->getToken('57'))) ) {
                    $alt2=1;
                }


                switch ($alt2) {
            	case 1 :
            	    // Poly.g:23:7: '0' .. '9' 
            	    {
            	    $this->matchRange(48,57); 

            	    }
            	    break;

            	default :
            	    if ( $cnt2 >= 1 ) break 2;//loop2;
                        $eee =
                            new EarlyExitException(2, $this->input);
                        throw $eee;
                }
                $cnt2++;
            } while (true);


            }

            $this->state->type = $_type;
            $this->state->channel = $_channel;
        }
        catch(Exception $e){
            throw $e;
        }
    }
    // $ANTLR end "INT"

    // $ANTLR start "WS"
    function mWS(){
        try {
            $_type = PolyLexer::$WS;
            $_channel = PolyLexer::$DEFAULT_TOKEN_CHANNEL;
            // Poly.g:25:4: ( ( ' ' | '\\t' | '\\r' | '\\n' )+ ) 
            // Poly.g:25:6: ( ' ' | '\\t' | '\\r' | '\\n' )+ 
            {
            // Poly.g:25:6: ( ' ' | '\\t' | '\\r' | '\\n' )+ 
            $cnt3=0;
            //loop3:
            do {
                $alt3=2;
                $LA3_0 = $this->input->LA(1);

                if ( (($LA3_0>=$this->getToken('9') && $LA3_0<=$this->getToken('10'))||$LA3_0==$this->getToken('13')||$LA3_0==$this->getToken('32')) ) {
                    $alt3=1;
                }


                switch ($alt3) {
            	case 1 :
            	    // Poly.g: 
            	    {
            	    if ( ($this->input->LA(1)>=$this->getToken('9') && $this->input->LA(1)<=$this->getToken('10'))||$this->input->LA(1)==13||$this->input->LA(1)==32 ) {
            	        $this->input->consume();

            	    }
            	    else {
            	        $mse = new MismatchedSetException(null,$this->input);
            	        $this->recover($mse);
            	        throw $mse;}


            	    }
            	    break;

            	default :
            	    if ( $cnt3 >= 1 ) break 2;//loop3;
                        $eee =
                            new EarlyExitException(3, $this->input);
                        throw $eee;
                }
                $cnt3++;
            } while (true);

              skip();

            }

            $this->state->type = $_type;
            $this->state->channel = $_channel;
        }
        catch(Exception $e){
            throw $e;
        }
    }
    // $ANTLR end "WS"

    function mTokens(){
        // Poly.g:1:8: ( T__8 | T__9 | ID | INT | WS ) 
        $alt4=5;
        $LA4 = $this->input->LA(1);
        if($this->getToken('43')== $LA4)
            {
            $alt4=1;
            }
        else if($this->getToken('94')== $LA4)
            {
            $alt4=2;
            }
        else if($this->getToken('97')== $LA4||$this->getToken('98')== $LA4||$this->getToken('99')== $LA4||$this->getToken('100')== $LA4||$this->getToken('101')== $LA4||$this->getToken('102')== $LA4||$this->getToken('103')== $LA4||$this->getToken('104')== $LA4||$this->getToken('105')== $LA4||$this->getToken('106')== $LA4||$this->getToken('107')== $LA4||$this->getToken('108')== $LA4||$this->getToken('109')== $LA4||$this->getToken('110')== $LA4||$this->getToken('111')== $LA4||$this->getToken('112')== $LA4||$this->getToken('113')== $LA4||$this->getToken('114')== $LA4||$this->getToken('115')== $LA4||$this->getToken('116')== $LA4||$this->getToken('117')== $LA4||$this->getToken('118')== $LA4||$this->getToken('119')== $LA4||$this->getToken('120')== $LA4||$this->getToken('121')== $LA4||$this->getToken('122')== $LA4)
            {
            $alt4=3;
            }
        else if($this->getToken('48')== $LA4||$this->getToken('49')== $LA4||$this->getToken('50')== $LA4||$this->getToken('51')== $LA4||$this->getToken('52')== $LA4||$this->getToken('53')== $LA4||$this->getToken('54')== $LA4||$this->getToken('55')== $LA4||$this->getToken('56')== $LA4||$this->getToken('57')== $LA4)
            {
            $alt4=4;
            }
        else if($this->getToken('9')== $LA4||$this->getToken('10')== $LA4||$this->getToken('13')== $LA4||$this->getToken('32')== $LA4)
            {
            $alt4=5;
            }
        else{
            $nvae =
                new NoViableAltException("", 4, 0, $this->input);

            throw $nvae;
        }

        switch ($alt4) {
            case 1 :
                // Poly.g:1:10: T__8 
                {
                $this->mT__8(); 

                }
                break;
            case 2 :
                // Poly.g:1:15: T__9 
                {
                $this->mT__9(); 

                }
                break;
            case 3 :
                // Poly.g:1:20: ID 
                {
                $this->mID(); 

                }
                break;
            case 4 :
                // Poly.g:1:23: INT 
                {
                $this->mINT(); 

                }
                break;
            case 5 :
                // Poly.g:1:27: WS 
                {
                $this->mWS(); 

                }
                break;

        }

    }



}
?>

File: examples/polydiff/PolyLexer.java
// $ANTLR 3.1.3 ??? 05, 2009 23:29:28 Poly.g 2009-05-06 00:12:20

import org.antlr.runtime.*;
import java.util.Stack;
import java.util.List;
import java.util.ArrayList;

public class PolyLexer extends Lexer {
    public static final int WS=7;
    public static final int INT=5;
    public static final int MULT=4;
    public static final int ID=6;
    public static final int EOF=-1;
    public static final int T__9=9;
    public static final int T__8=8;

    // delegates
    // delegators

    public PolyLexer() {;} 
    public PolyLexer(CharStream input) {
        this(input, new RecognizerSharedState());
    }
    public PolyLexer(CharStream input, RecognizerSharedState state) {
        super(input,state);

    }
    public String getGrammarFileName() { return "Poly.g"; }

    // $ANTLR start "T__8"
    public final void mT__8() throws RecognitionException {
        try {
            int _type = T__8;
            int _channel = DEFAULT_TOKEN_CHANNEL;
            // Poly.g:3:6: ( '+' )
            // Poly.g:3:8: '+'
            {
            match('+'); 

            }

            state.type = _type;
            state.channel = _channel;
        }
        finally {
        }
    }
    // $ANTLR end "T__8"

    // $ANTLR start "T__9"
    public final void mT__9() throws RecognitionException {
        try {
            int _type = T__9;
            int _channel = DEFAULT_TOKEN_CHANNEL;
            // Poly.g:4:6: ( '^' )
            // Poly.g:4:8: '^'
            {
            match('^'); 

            }

            state.type = _type;
            state.channel = _channel;
        }
        finally {
        }
    }
    // $ANTLR end "T__9"

    // $ANTLR start "ID"
    public final void mID() throws RecognitionException {
        try {
            int _type = ID;
            int _channel = DEFAULT_TOKEN_CHANNEL;
            // Poly.g:18:4: ( ( 'a' .. 'z' )+ )
            // Poly.g:18:6: ( 'a' .. 'z' )+
            {
            // Poly.g:18:6: ( 'a' .. 'z' )+
            int cnt1=0;
            loop1:
            do {
                int alt1=2;
                int LA1_0 = input.LA(1);

                if ( ((LA1_0>='a' && LA1_0<='z')) ) {
                    alt1=1;
                }


                switch (alt1) {
            	case 1 :
            	    // Poly.g:18:6: 'a' .. 'z'
            	    {
            	    matchRange('a','z'); 

            	    }
            	    break;

            	default :
            	    if ( cnt1 >= 1 ) break loop1;
                        EarlyExitException eee =
                            new EarlyExitException(1, input);
                        throw eee;
                }
                cnt1++;
            } while (true);


            }

            state.type = _type;
            state.channel = _channel;
        }
        finally {
        }
    }
    // $ANTLR end "ID"

    // $ANTLR start "INT"
    public final void mINT() throws RecognitionException {
        try {
            int _type = INT;
            int _channel = DEFAULT_TOKEN_CHANNEL;
            // Poly.g:20:5: ( ( '0' .. '9' )+ )
            // Poly.g:20:7: ( '0' .. '9' )+
            {
            // Poly.g:20:7: ( '0' .. '9' )+
            int cnt2=0;
            loop2:
            do {
                int alt2=2;
                int LA2_0 = input.LA(1);

                if ( ((LA2_0>='0' && LA2_0<='9')) ) {
                    alt2=1;
                }


                switch (alt2) {
            	case 1 :
            	    // Poly.g:20:7: '0' .. '9'
            	    {
            	    matchRange('0','9'); 

            	    }
            	    break;

            	default :
            	    if ( cnt2 >= 1 ) break loop2;
                        EarlyExitException eee =
                            new EarlyExitException(2, input);
                        throw eee;
                }
                cnt2++;
            } while (true);


            }

            state.type = _type;
            state.channel = _channel;
        }
        finally {
        }
    }
    // $ANTLR end "INT"

    // $ANTLR start "WS"
    public final void mWS() throws RecognitionException {
        try {
            int _type = WS;
            int _channel = DEFAULT_TOKEN_CHANNEL;
            // Poly.g:22:4: ( ( ' ' | '\\t' | '\\r' | '\\n' )+ )
            // Poly.g:22:6: ( ' ' | '\\t' | '\\r' | '\\n' )+
            {
            // Poly.g:22:6: ( ' ' | '\\t' | '\\r' | '\\n' )+
            int cnt3=0;
            loop3:
            do {
                int alt3=2;
                int LA3_0 = input.LA(1);

                if ( ((LA3_0>='\t' && LA3_0<='\n')||LA3_0=='\r'||LA3_0==' ') ) {
                    alt3=1;
                }


                switch (alt3) {
            	case 1 :
            	    // Poly.g:
            	    {
            	    if ( (input.LA(1)>='\t' && input.LA(1)<='\n')||input.LA(1)=='\r'||input.LA(1)==' ' ) {
            	        input.consume();

            	    }
            	    else {
            	        MismatchedSetException mse = new MismatchedSetException(null,input);
            	        recover(mse);
            	        throw mse;}


            	    }
            	    break;

            	default :
            	    if ( cnt3 >= 1 ) break loop3;
                        EarlyExitException eee =
                            new EarlyExitException(3, input);
                        throw eee;
                }
                cnt3++;
            } while (true);

            skip();

            }

            state.type = _type;
            state.channel = _channel;
        }
        finally {
        }
    }
    // $ANTLR end "WS"

    public void mTokens() throws RecognitionException {
        // Poly.g:1:8: ( T__8 | T__9 | ID | INT | WS )
        int alt4=5;
        switch ( input.LA(1) ) {
        case '+':
            {
            alt4=1;
            }
            break;
        case '^':
            {
            alt4=2;
            }
            break;
        case 'a':
        case 'b':
        case 'c':
        case 'd':
        case 'e':
        case 'f':
        case 'g':
        case 'h':
        case 'i':
        case 'j':
        case 'k':
        case 'l':
        case 'm':
        case 'n':
        case 'o':
        case 'p':
        case 'q':
        case 'r':
        case 's':
        case 't':
        case 'u':
        case 'v':
        case 'w':
        case 'x':
        case 'y':
        case 'z':
            {
            alt4=3;
            }
            break;
        case '0':
        case '1':
        case '2':
        case '3':
        case '4':
        case '5':
        case '6':
        case '7':
        case '8':
        case '9':
            {
            alt4=4;
            }
            break;
        case '\t':
        case '\n':
        case '\r':
        case ' ':
            {
            alt4=5;
            }
            break;
        default:
            NoViableAltException nvae =
                new NoViableAltException("", 4, 0, input);

            throw nvae;
        }

        switch (alt4) {
            case 1 :
                // Poly.g:1:10: T__8
                {
                mT__8(); 

                }
                break;
            case 2 :
                // Poly.g:1:15: T__9
                {
                mT__9(); 

                }
                break;
            case 3 :
                // Poly.g:1:20: ID
                {
                mID(); 

                }
                break;
            case 4 :
                // Poly.g:1:23: INT
                {
                mINT(); 

                }
                break;
            case 5 :
                // Poly.g:1:27: WS
                {
                mWS(); 

                }
                break;

        }

    }


 

}